src/updatechecker.cpp

```diff
@@ -254,32 +289,65 @@ void UpdateChecker::PerformUpdateCheck(bool manual)
             }
         }
 
+        const std::string currentVersion = WideToAnsi(Settings::GetAppBuildVersion());
+
+        Settings::WriteConfigValue("LastCheckTime", time(NULL));
+
         if (!appcast.ReleaseNotesURL.empty())
             CheckForInsecureURL(appcast.ReleaseNotesURL, "release notes");
         if (!appcast.DownloadURL.empty())
             CheckForInsecureURL(appcast.DownloadURL, "update file");
 
-        Settings::WriteConfigValue("LastCheckTime", time(NULL));
+        if (appcast.SilentInstall)
+        {
+            // Check if our version is out of date.
+            if (!appcast.IsValid() || CompareVersions(currentVersion, appcast.Version) >= 0)
+            {
+                // The same or newer version is already installed.
+                return;
+            }
 
-        const std::string currentVersion =
-                WideToAnsi(Settings::GetAppBuildVersion());
+            if (appcast.DownloadURL != "")
+            {
+                const std::wstring tmpdir = winsparkle::CreateUniqueTempDirectory();
+                Settings::WriteConfigValue("UpdateTempDir", tmpdir);
 
-        // Check if our version is out of date.
-        if ( !appcast.IsValid() || CompareVersions(currentVersion, appcast.Version) >= 0 )
-        {
-            // The same or newer version is already installed.
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
-        }
+                UpdateDownloadSink sink(*this, tmpdir);
+                DownloadFile(appcast.DownloadURL, &sink, this);
+                sink.Close();
 
-        // Check if the user opted to ignore this particular version.
-        if ( ShouldSkipUpdate(appcast) )
-        {
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
+                if (Settings::HasDSAPubKeyPem())
+                {
+                    SignatureVerifier::VerifyDSASHA1SignatureValid(sink.GetFilePath(), appcast.DsaSignature);
+                }
```

Comment:
Shouldn't there be an action of some kind if the signature check fails to block the update?

Reply:
With silent updates it is probably more likely that a malicious install would go undetected. The SHA value is more of an integrity test than a security feature as a malicious actor could fame that as well. So I am not certain how far we really need to go with this.
This topic is probably of interest to the security team so they are probably better suited to answer.

Reply:
Good idea, I will solicit the security team for feedback.
@andrewkeller I'm curious if you added any additional security when you added silent updates for Mac? For Windows we're downloading the update installer using the client api.